### PR TITLE
[Server][DBTableMonitoring] Add a update processing of hostname.

### PR DESCRIPTION
**Pull request by Mnakagawa**

### CODE REVIEW

File: server/src/DBTablesMonitoring.cc [hunk 1]
```diff
@@ -2101,8 +2099,16 @@ void DBTablesMonitoring::updateHosts(const HostInfoList &hostInfoList,
...
+		if (currHostItr != currValidHosts.end()) {
+			const HostInfo &existHost = *currHostItr->second;
+			if (existHost.hostName != newHostInfo.hostName){
+				updatedHostInfoList.push_back(newHostInfo);
+				continue;
+			}
+		}
 		if (currValidHosts.erase(newHostInfo.id) >= 1) {
-			// The host already exits. We have nrothing to do.
+			// The host already exits. We have to do nothing.
```
Comment (kz0817) [inline server/src/DBTablesMonitoring.cc]: Please fix the comment in another commit.

File: server/src/DBTablesMonitoring.cc [hunk 2]
```diff
@@ -2101,8 +2099,16 @@ void DBTablesMonitoring::updateHosts(const HostInfoList &hostInfoList,
...
 		const HostInfo &newHostInfo = *newHostsItr;
+		HostIdHostInfoMapConstIterator currHostItr = currValidHosts.find(newHostInfo.id);
+		if (currHostItr != currValidHosts.end()) {
+			const HostInfo &existHost = *currHostItr->second;
+			if (existHost.hostName != newHostInfo.hostName){
+				updatedHostInfoList.push_back(newHostInfo);
+				continue;
+			}
+		}
 		if (currValidHosts.erase(newHostInfo.id) >= 1) {
```
Comment (kz0817) [inline server/src/DBTablesMonitoring.cc]: This code does the search twice (find() and erase()) when an element in currValidHosts is removed.
erase() accepts an iterator that is directly pointing an element.

I.e.
currHostItr should be passed to erase() when it should be removed.

**Comment by kz0817** [comment]
Please add a unit test to check the feature.

**Comment by Mnakagawa** [comment]
I create a new pull request.
Please, see that.

**Pull request closed without merging**